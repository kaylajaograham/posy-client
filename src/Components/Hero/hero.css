

.hero {
    /* Sizing */
    width: 100vw;
    height: 100vh;
    
    /* Text styles */
    text-align: center;

    
    /* Background styles */
    background-image: url(https://i.pinimg.com/originals/a6/d0/54/a6d0549948c9b9b071b2234f5390c973.jpg);
    background-size: cover;
    background-position: center center;
    background-repeat: no-repeat;
    background-attachment: fixed;
    margin-bottom: 30px;
    opacity: 90%;
}

.herotext {
    background: none;
    padding: 150px 30px;

}

.heroTitle {
    font-family: Verdana;
    letter-spacing: 20px;
    font-size: 38px;
    color: white;
    padding: 50px 20px;
    text-shadow: 3px 3px 3px black;
    background: none;
    margin-top: 0px;

}

.heroTag {
    font-family: sans-serif;
    letter-spacing: 5px;
    font-size: 22px;
    color: white;
    padding: 0 20px;
    text-shadow: 3px 3px 3px black;
    margin-bottom: 50px;
    background: none;

}

.herolinks {
    border-radius: 15px;
    padding: 8px 15px;
    text-decoration: none;
    font-weight: bold;
    color: white;
    border-bottom: 1px solid white; 
    margin: 50px;
    background-color: #21633E;

}

/* NAV CSS */
.navbar {
    width: 100%;
    list-style: none;
    border-bottom: 1px solid #ccc; 
    border-top: 1px solid #ccc; 
}

.navlinks {
    display: inline-block;
    padding: 8px;
    text-decoration: none;
    font-weight: bold;
    color: #21633E;
    border-right: 1px solid #ccc; 
}

ul {
    padding: 0;
}

/* ABOUT CSSSSSS */


.aboutInner1, .aboutInner2, .aboutInner3 {
    /* Sizing */
    width: 100vw;
    height: 40vh;
    
    /* Flexbox stuff */
    display: flex;
    justify-content: center;
    align-items: center;
    
    /* Text styles */
    text-align: center;
    
    /* Background styles */
    background-size: cover;
    background-position: center center;
    background-repeat: no-repeat;
    background-attachment: fixed;

    border-radius: 25px;
    margin-top: 50px;
 
}

.aboutTitle, .aboutTag {
    background: none;
    font-family: Verdana, Geneva, Tahoma, sans-serif;
    color: white;
    letter-spacing: 6px;
    font-size: 26px;
    text-shadow: 3px 3px 3px black;
}

.aboutInner2 {
    background-image: url(https://images.unsplash.com/photo-1531762834015-57aa765c6047?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80);
}

.aboutInner1 {
    background-image: url(https://www.nationalgeographic.com/content/dam/news/photos/000/607/60780.jpg);
}

.aboutInner3 {
    background-image: url(https://wordpress.accuweather.com/wp-content/uploads/2019/02/snow-in-pinehurst-seattle-washington-2-9-19.31.5620AM.png);
}



