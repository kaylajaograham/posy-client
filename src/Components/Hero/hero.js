import React, {Component} from 'react';
import './hero.css';
import { Link } from 'react-router-dom';


class Hero extends Component {

    render() {
        return (
            <div className='motherHero'>
            
            <div className='navbar'>
              <ul>
                <Link className='navlinks' to='/'>Home</Link>
                <Link className='navlinks' to='/register'>Register</Link>
                <Link className='navlinks' to='/login'>Login</Link>              
              </ul>
            </div>
            
            <div className='hero'>
                <div className='herotext'>
                    <h1 className='heroTitle'>POSY</h1>
                    <h2 className='heroTag'>positivity in your pocket</h2>
                </div>
                <Link className='herolinks' to='/register'>Register</Link>
            </div>

            <div className='about'>
                <div className='aboutInner1'>
                    <h4 className='aboutTag'>Let your thoughts flow</h4>
                </div>

                <div className='aboutInner2'>
                    <h4 className='aboutTag'>Clear your mind</h4>
                </div>

                <div className='aboutInner3'>
                    <h4 className='aboutTag'>Create healthy habits</h4>
                </div>
            </div>

        </div>
        )
    }
} 

export default Hero;

