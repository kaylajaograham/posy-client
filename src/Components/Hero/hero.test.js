import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Hero from './hero';

describe(`Hero component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  };

  it('renders a Hero by default', () => {
    const wrapper = shallow(<Hero />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders the Landing given props', () => {
    const wrapper = shallow(<Hero {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})