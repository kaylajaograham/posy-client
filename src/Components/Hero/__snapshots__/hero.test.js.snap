// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component renders a Hero by default 1`] = `
<div
  className="motherHero"
>
  <div
    className="navbar"
  >
    <Nav />
  </div>
  <div
    className="hero"
  >
    <div
      className="herotext"
    >
      <h1
        className="heroTitle"
      >
        POSY
      </h1>
      <h2
        className="heroTag"
      >
        positivity in your pocket
      </h2>
    </div>
    <Link
      className="herolinks"
      to="/register"
    >
      Register
    </Link>
  </div>
  <div
    className="about"
  >
    <div
      className="aboutInner1"
    >
      <h4
        className="aboutTag"
      >
        Let your thoughts flow
      </h4>
    </div>
    <div
      className="aboutInner2"
    >
      <h4
        className="aboutTag"
      >
        Clear your mind
      </h4>
    </div>
    <div
      className="aboutInner3"
    >
      <h4
        className="aboutTag"
      >
        Create healthy habits
      </h4>
    </div>
  </div>
</div>
`;

exports[`Hero component renders the Landing given props 1`] = `
<div
  className="motherHero"
>
  <div
    className="navbar"
  >
    <Nav />
  </div>
  <div
    className="hero"
  >
    <div
      className="herotext"
    >
      <h1
        className="heroTitle"
      >
        POSY
      </h1>
      <h2
        className="heroTag"
      >
        positivity in your pocket
      </h2>
    </div>
    <Link
      className="herolinks"
      to="/register"
    >
      Register
    </Link>
  </div>
  <div
    className="about"
  >
    <div
      className="aboutInner1"
    >
      <h4
        className="aboutTag"
      >
        Let your thoughts flow
      </h4>
    </div>
    <div
      className="aboutInner2"
    >
      <h4
        className="aboutTag"
      >
        Clear your mind
      </h4>
    </div>
    <div
      className="aboutInner3"
    >
      <h4
        className="aboutTag"
      >
        Create healthy habits
      </h4>
    </div>
  </div>
</div>
`;
